File: javascript/easy/1365.smallerNumbersThanCurrent.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function (nums) {
  const resArr = [];

  for (let i = 0; i < nums.length; i++) {
    let counter = 0;
    for (let j = 0; j < nums.length; j++) {
      if (nums[i] > nums[j] && nums[i] != nums[j]) {
        counter++;
      }
    }
    resArr.push(counter);
  }

  return resArr;
};

console.log(smallerNumbersThanCurrent([8, 1, 2, 2, 3])); // [4,0,1,1,3]
// Explanation:
// For nums[0]=8 there exist four smaller numbers than it (1, 2, 2 and 3).
// For nums[1]=1 does not exist any smaller number than it.
// For nums[2]=2 there exist one smaller number than it (1).
// For nums[3]=2 there exist one smaller number than it (1).
// For nums[4]=3 there exist three smaller numbers than it (1, 2 and 2).

console.log(smallerNumbersThanCurrent([6, 5, 4, 8])); // [2,1,0,3]
console.log(smallerNumbersThanCurrent([7, 7, 7, 7])); // [0,0,0,0]

File: python/easy/1365.smallerNumbersThanCurrent.py
from typing import List


class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        res_list = []

        for i in nums:
            counter = 0
            for j in nums:
                if i > j and i != j:
                    counter += 1
            res_list.append(counter)

        return res_list


if __name__ == "__main__":
    solution = Solution()
    print(solution.smallerNumbersThanCurrent([8, 1, 2, 2, 3]))  # [4,0,1,1,3]
    print(solution.smallerNumbersThanCurrent([6, 5, 4, 8]))  # [2,1,0,3]
    print(solution.smallerNumbersThanCurrent([7, 7, 7, 7]))  # [0,0,0,0]
